weather-oracle/sources/weather.move

```diff
@@ -0,0 +1,377 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// This module defines a weather oracle that can post weather updates for major cities around the world.
+module oracle::weather {
+    use std::option::{Self, Option};
+    use std::string::{Self, String};
+
+    use sui::dynamic_object_field as dof;
+    use sui::object::{Self, UID};
+    use sui::package;
+    use sui::transfer::{Self};
+    use sui::tx_context::{Self, TxContext};
+
+    /// Define a capability for the admin of the oracle.
+    struct AdminCap has key, store { id: UID }
+
+    /// // Define a one-time witness to create the `Publisher` of the oracle.
+    struct WEATHER has drop {}
+
+    // Define a struct for the weather oracle
+    struct WeatherOracle has key {
+        id: UID,
+        /// The address of the oracle.
+        address: address,
```

Comment:
Not certain where this address is needed?

Reply:
I concur, but it is a matter of design choice. He followed the example of this: https://suiexplorer.com/object/0xaa14ecfe646b35f118332203df2013bd5c1edd7ef8547d2c85f64b2e15d58279 , which also has the address as a property, so that is why we included it.